### PR TITLE
Add dashboard template

### PR DESCRIPTION
**Pull request by SicongLiuSimon**
Add dashboard template to the extension generator, with homepage action, dashboard tab, dashboard insight, dashboard toolbar and navigation section.

### CODE REVIEW

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+pool:
+  vmImage: 'ubuntu-16.04'
+
+steps:
+- task: NodeTool@0
```
Comment (llali) [inline .azure-pipelines.yml:5]: this is part of dashboard change?

File: .github/assignment.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+    perform: true,
+    assignees: [ aeschli ]
+}
```
Comment (llali) [inline .github/assignment.yml:4]: we should not add .github files

File: .gitignore [hunk 3]
```diff
@@ -1,3 +1,2 @@
 /node_modules
-.DS_Store
-npm-debug.log
```
Comment (llali) [inline .gitignore:3]: I think we should revert this?

File: generators/app/env.js [hunk 4]
```diff
@@ -2,29 +2,28 @@
...
+    }
+    return fallbackVersion;
 });
 
-module.exports.getLatestVSCodeVersion = function() { return vscodePromise; };
-
-var azdataFallbackVersion = '*';
-module.exports.azdataVersion = azdataFallbackVersion;
+module.exports.getLatestVSCodeVersion = function() { return promise; };
+var azdataFallbackVersion = '*';// {{ADS EDIT}}
```
Comment (llali) [inline generators/app/env.js:28]: >;// {{ADS EDIT}} [](start = 31, length = 16)

can you please add ;// {{ADS EDIT}} here?

File: generators/app/grammarConverter.js [hunk 5]
```diff
@@ -51,62 +49,76 @@ function convertGrammar(location, extensionConfig) {
...
-        if (Array.isArray(languageInfo.fileTypes)) {
-            extensionConfig.languageExtensions = languageInfo.fileTypes.map(function (ft) { return '.' + ft; });
-        } else {
-            extensionConfig.languageExtensions = languageId ? ['.' + languageId] : [];
-        }
+    // evaluate file extensions
+    if (Array.isArray(languageInfo.fileTypes)) {
+        extensionConfig.languageExtensions = languageInfo.fileTypes.map(function (ft) { return '.' + ft; });
+    } else {
+        extensionConfig.languageExtensions = languageId ? ['.' + languageId] : [];
```
Comment (llali) [inline generators/app/grammarConverter.js:118]: there are too many changes in this file. are they related to dashboard?

File: generators/app/index.js [hunk 6]
```diff
@@ -77,13 +78,17 @@ module.exports = class extends Generator {
                         value: 'ext-command-js'
                     },
                     {
-                        name: 'New Dashboard Insight',
-                        value: 'ext-insight'
```
Comment (llali) [inline generators/app/index.js:81]: are we removing dashboard insight and replacing with new dashboard template?

File: generators/app/index.js [hunk 7]
```diff
@@ -204,6 +211,7 @@ module.exports = class extends Generator {
                 return snippetPrompt();
             },
 
+
             askForLocalizationLanguageId: () => {
```
Comment (llali) [inline generators/app/index.js:215]: remove extra line

File: generators/app/index.js [hunk 8]
```diff
@@ -116,21 +121,20 @@ module.exports = class extends Generator {
...
+                            name: 'Yes, import an existing theme but keep it as tmTheme file.',
+                            value: 'import-keep'
+                        },
+                        {
+                            name: 'Yes, import an existing theme and inline it in the Visual Studio Code color theme file.',
+                            value: 'import-inline'
+                        }
                     ]
                 }).then(answer => {
-                    let inline = true;
```
Comment (llali) [inline generators/app/index.js:133]: why removing this line?

File: generators/app/index.js [hunk 9]
```diff
@@ -222,53 +230,211 @@ module.exports = class extends Generator {
...
+            },
+
+            askForServerTab: () => {// {{ADS EDIT}}
+                if (generator.extensionConfig.type !== 'ext-dashboard' || !generator.extensionConfig.addDashboardTab) {// {{ADS EDIT}}
+                    return Promise.resolve();// {{ADS EDIT}}
+                }
+
+                generator.extensionConfig.isCustomization = true;// {{ADS EDIT}}
+
+                return generator.prompt({// {{ADS EDIT}}
```
Comment (llali) [inline generators/app/index.js:323]: you don't need to put ADS Edit in every line. you can put it on top of a block or around something. if it's one line of inside a big VS code block then maybe put it in line

File: generators/app/index.js [hunk 10]
```diff
@@ -222,53 +230,211 @@ module.exports = class extends Generator {
...
+                    default: true// {{ADS EDIT}}
+                }).then(function (answer) {// {{ADS EDIT}}
+                    generator.extensionConfig.addNavSection = answer.addNavSection;// {{ADS EDIT}}
+                });
+            },
+
+
+
+
             // Ask for extension display name ("displayName" in package.json)
```
Comment (llali) [inline generators/app/index.js:438]: remove extra lines above

File: generators/app/index.js [hunk 11]
```diff
@@ -383,7 +506,7 @@ module.exports = class extends Generator {
                     name: 'checkJavaScript',
                     message: 'Enable JavaScript type checking in \'jsconfig.json\'?',
                     default: false
-                }).then(function (strictJavaScriptAnswer) {
+                }).then(strictJavaScriptAnswer => {
```
Comment (llali) [inline generators/app/index.js:509]: they are too many changes in this PR. makes it hard to review. please make sure the changes are only for dashboard. If they are other changes please send separate PRs. in general sending smaller PRs makes it easier to review them

File: generators/app/templates/ext-dashboard/src/controllers/mainController.js [hunk 12]
```diff
@@ -0,0 +1,80 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
```
Comment (llali) [inline generators/app/templates/ext-dashboard/src/controllers/mainController.js:4]: please put a line after copyright

File: generators/app/index.js (hunk 6 above)
Comment (SicongLiuSimon) [reply generators/app/index.js]: Yes, I think so, since dashboard template does more than just insight
